service/storage_proxy.cc

```diff
@@ -5388,6 +5400,25 @@ future<> storage_proxy::wait_for_hints_to_be_replayed(utils::UUID operation_id,
     co_return;
 }
 
+void storage_proxy::mark_responded(const gms::inet_address& ep, clock_type::duration rtt) {
+    last_seen_info& info = _last_seen[ep];
```

Comment:
We really need a node_ptr to avoid all those hash tables everywhere indexing the same thing.

Please provide before/after perf_simple_query instruction count to see how much this costs.

Reply:
https://github.com/scylladb/scylla/pull/8393#issuecomment-834477194